src/track.rs

```diff
@@ -155,10 +159,14 @@ impl Client {
             ..Default::default()
         }).await.ok()?.into_inner().status?;
 
+        let mut labels = s.labels.clone();
+
```

Comment:
There's no need to clone the labels map since we already own the Status struct, just need to make `s` mutable to call `s.labels.remove()`, or use HashMap.get() and clone the result since the difference in performance doesn't matter here.